linkedlist/singly/go/dll.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+)
+
+// DllNode is an element in the linked list
+type DllNode struct {
+	element interface{}
+	prev    *DllNode
+	next    *DllNode
+}
+
+func (n *DllNode) String() string {
```

Comment:
What's this method for? @chidiwilliams

Reply:
That's the string representation, for printing out the value.

Wanted to illustrate the previous and next pointers. See: https://github.com/oyekanmiayo/data-structures-all-langs/pull/25/files/954b94d61b99cb792740a33e0b042864e7f04785#diff-a84ab3123aca5dc7dbc36a25fb694844696c67a8425bb02c4b9e41165281d6fdR213

I'll add a comment on the method.

Reply:
That's the string representation. `fmt.Print()` uses it to print out the value.

The logic in there is because I wanted to illustrate the previous and next pointers. See: https://github.com/oyekanmiayo/data-structures-all-langs/pull/25/files/954b94d61b99cb792740a33e0b042864e7f04785#diff-a84ab3123aca5dc7dbc36a25fb694844696c67a8425bb02c4b9e41165281d6fdR213

Reply:
Oh, I see. Not very sure this way of presenting is immediately intuitive.

What do you think about this `1<->3<->2<->nil` instead of `(nil)1(3)<->(1)3(2)<->(3)2(nil)`? @chidiwilliams 

*the last nil can be omitted for brevity*

Reply:
Sure, that works too. I saw it the other language implementations.

Why I suggested this was that the arrows alone don't show that the nodes contain the `prev` pointer. You could put the arrows on the singly-linked list and it would look exactly the same.

But I agree that the first one looks cleaner. I'll change it back.

Reply:
Oh, so singly linkedlist arrows should look like `1 ->3 ->2 -> nil` @chidiwilliams

Reply:
Fixed